app/js/lib/foxbox/services/light.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+import BaseService from './base';
+
+const TYPE = 'Light';
+
+const p = Object.freeze({
+  // Public getter only properties.
+  manufacturer: Symbol('manufacturer'),
+  model: Symbol('model'),
+  name: Symbol('name'),
+
+  // Private getter only properties.
+  adapter: Symbol('adapter'),
+});
+
+export default class LightService extends BaseService {
+  constructor(props, config) {
+    super(props, config);
+
+    // Public getter only properties.
+    this[p.manufacturer] = props.properties.manufacturer;
+    this[p.model] = props.properties.model;
+    this[p.name] = props.properties.name;
+
+    // Private getter only properties.
+    this[p.adapter] = props.adapter;
+
+    Object.seal(this);
+  }
+
+  // Public getter only properties.
+  get type() {
+    return TYPE;
+  }
+
+  get manufacturer() {
+    return this[p.manufacturer];
+  }
+
+  get model() {
+    return this[p.model];
+  }
+
+  get name() {
+    return this[p.name];
+  }
+
+  // Public dynamic property getters.
+  isAvailable() {
+    return this.get('available')
+      .then((response) => {
+        for (let id in response) {
```

Comment:
__nit:__ maybe ```Object.keys().forEach``` will be safer since we don't check that the property is own one? I'd even force [1] :)

[1] http://eslint.org/docs/rules/guard-for-in